### PR TITLE
Add editor for track elements.

### PR DESCRIPTION
**Pull request by tomprince**
This is an editor for #72.

### CODE REVIEW

File: objects/Editors/script.lua [hunk 1]
```diff
@@ -66,7 +67,7 @@ function toggleObjects()
                 guid = bagObject.guid,
                 position = contents[bagObject.guid],
                 rotation = {0,180,0},
-                callback_function = function(obj) obj.setLock(true) objectsMoved = objectsMoved + 1 end,
+                callback_function = function(obj) obj.setLock(true) objectsMoved = objectsMoved + 1 obj.setPosition(contents[bagObject.guid]) end,
```
Comment (tomprince) [inline objects/Editors/script.lua:70]: I'm not entirely sure why this is necessary, but without it, the element bag seems to get displaced to a randomish location either above or below the table.

**Comment by iakona** [comment]
so i just swapped out element tokens for new models earlier so i think this may need an update

File: objects/Editors/contained/SpiritEditor/script.lua [hunk 2]
```diff
@@ -12,24 +13,32 @@ function onLoad()
...
 end
 
 function scan()
     local objs = upCast(self, 0.4, 0.1, {"Tile"})
-    if #objs == 0 then
-        clearButtons()
-        return
+    local spiritBoard
+    for _, hit in pairs(objs) do
+        if hit.getVar("elements") ~= nil or hit.hasTag("Any") then
```
Comment (iakona) [inline objects/Editors/contained/SpiritEditor/script.lua:25]: elements are now the "Generic" type so they shouldn't be being returned here anymore

File: objects/Editors/contained/SpiritEditor/script.lua [hunk 3]
```diff
@@ -72,6 +81,97 @@ function createButtons(obj)
...
+
+local function round(val, quantum)
+    return math.floor(val/quantum+0.5)*quantum
+end
+
+function updateElements(player)
+    if currentSpirit == nil then
+        return
+    end
+    local hits = upCast(currentSpirit, 0.4, 0.1, {"Tile"})
```
Comment (iakona) [inline objects/Editors/contained/SpiritEditor/script.lua:119]: elements are now type "Generic"

**Comment by iakona** [comment]
I'll just make the changes on my end

**Pull request merged**